sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,577 @@
+        and returns either TOR A or TOR B. Note that this API should return mux-direction
+        regardless of whether the link is active/routing state or not.
+
+        Args:
+            port:
+                an Integer, the actual physical port connected to a Y cable
+
+        Returns:
+            an Integer, 1 if the mux is pointing to TOR A .
+                      , 2 if the mux is pointing to TOR B.
+                      , -1 if checking which side mux is pointing to API fails.
+        """
+
+        raise NotImplementedError
+
+    def check_active_linked_tor_side(self, port):
+        """
+        This API specifically checks which side of the Y cable is actively linked and routing
+        and returns either TOR A or TOR B.
+
+        Args:
+            port:
+                an Integer, the actual physical port connected to a Y cable
+
+        Returns:
+            an Integer, 1 if TOR A is actively linked and routing(TOR 1).
+                      , 2 if TOR B is actively linked and routing(TOR 2).
+                      , 0 if nothing linked and actively routing
+                      , -1 if checking which side linked for routing API fails.
+        """
+
+        raise NotImplementedError
+
+    def check_if_link_is_active_for_NIC(self, port):
+        """
+        This API specifically checks if NIC side of the Y cable's link is active
+
+        Args:
+            port:
+                 an Integer, the actual physical port connected to a Y cable
+
+        Returns:
+            a boolean, true if the link is active
+                     , false if the link is not active
+        """
+
+        raise NotImplementedError
+
+    def check_if_link_is_active_for_torA(self, port):
+        """
+        This API specifically checks if tor A side of the Y cable's link is active
+
+        Args:
+            port:
+                 an Integer, the actual physical port connected to a Y cable
+
+        Returns:
+            a boolean, true if the link is active
+                     , false if the link is not active
+        """
+
+        raise NotImplementedError
+
+    def check_if_link_is_active_for_torB(self, port):
+        """
+        This API specifically checks if tor B side of the Y cable's link is active
+
+        Args:
+            port:
+                 an Integer, the actual physical port connected to a Y cable
+
+        Returns:
+            a boolean, true if the link is active
+                     , false if the link is not active
+        """
+
+        raise NotImplementedError
+
+    def get_eye_info(self, port):
+        """
+        This API specifically returns the EYE height value for a specfic port.
+        The target could be local side, TOR1, TOR2, NIC etc.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+            target:
+                 an Integer, the target on which to enable the PRBS
```

Comment:
Maybe rather than explicitly stating "An Integer", we simply state "One of the following predefined constants:". This would also allow us to change the underlying type transparently if desired.

I would prefer using an Enum class here. However, there was a big change between Python 2 and 3, and sonic-platform-common still needs to support Python 2 for a little while longer, so I think we avoid Enums for now, and maybe change over after the package is Python 3-only.

Reply:
fixed